File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        operators = ["+", "-", "*", "/"]
        for i in tokens:
            if i.isnumeric() or (i[0] == "-" and i[1:].isnumeric()):
                stack.append(int(i))
            else:
                n = stack.pop()
                m = stack.pop()
                if i == "+":
                    stack.append(m + n)
                elif i == "-":
                    stack.append(m - n)
                elif i == "*":
                    stack.append(m * n)
                elif i == "/":
                    stack.append(int(m / n))
        
        return stack[0]

File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.js
/**
 * @param {string[]} tokens
 * @return {number}
 */
var evalRPN = function(tokens) {
    const stack = [];
    
    // Operator map for evaluation
    const operators = {
        "+": (a, b) => a + b,
        "-": (a, b) => a - b,
        "*": (a, b) => a * b,
        "/": (a, b) => a / b >= 0 ? Math.floor(a / b) : Math.ceil(a / b)
    };

    for (const token of tokens) {
        // Check if the token is an operator
        if (operators[token]) {
            const secondVal = stack.pop();
            const firstVal = stack.pop();
            const result = operators[token](firstVal, secondVal);
            stack.push(result); // Push the result back onto the stack
        } else {
            // Token is a number, push it onto the stack
            stack.push(parseInt(token, 10)); // Convert string to integer
        }
    }
    
    return stack.pop();

};